src/main/java/org/opensearch/ad/task/ADTaskManager.java

```diff
@@ -1972,4 +2048,153 @@ public synchronized void removeStaleRunningEntity(ADTask adTask, String entity,
         }
     }
 
+    /**
```

Comment:
Could you comment on what maintenance we actually do as you did for maintainRunningRealtimeTasks?

Reply:
Will add more comments